homeassistant/components/modbus/binary_sensor.py

```diff
@@ -23,21 +31,30 @@ async def async_setup_platform(
     discovery_info: DiscoveryInfoType | None = None,
 ) -> None:
     """Set up the Modbus binary sensors."""
-    sensors = []
 
     if discovery_info is None:  # pragma: no cover
         return
 
+    sensors: list[ModbusBinarySensor | SlaveSensor] = []
+    hub = get_hub(hass, discovery_info[CONF_NAME])
     for entry in discovery_info[CONF_BINARY_SENSORS]:
-        hub = get_hub(hass, discovery_info[CONF_NAME])
-        sensors.append(ModbusBinarySensor(hub, entry))
-
+        slaves = [SlaveSensor(i, entry) for i in range(1, entry[CONF_SLAVE_COUNT] + 1)]
+        sensors.append(ModbusBinarySensor(hub, entry, slaves))
```

Comment:
If there are slaves, why still add this sensor? Isn't this data already represented in the other entities?

Also, if this entity is disabeld, I guess the slaves won't work anymore? It might be better to use a data update coordinator in this case to make sure data is fetched once and given to all entities that are enabled.

Reply:
the sensor itself is bit0, and the slaves are bit1-n, so if we receive 0x05 on sensor "mySensor" defined with slave_count = 2

"mySensor" is STATE_ON
"mySensor_1" is STATE_OFF
"mySensor_2" is STATE_ON

I followed your suggestion and changed to use the update coordinator and that is a better solution !